go/common/entity/entity.go

```diff
@@ -51,10 +51,6 @@ type Entity struct { // nolint: maligned
 	// will sign the descriptor with the node signing key rather than the
 	// entity signing key.
 	Nodes []signature.PublicKey `json:"nodes,omitempty"`
-
-	// AllowEntitySignedNodes is true iff nodes belonging to this entity
-	// may be signed with the entity signing key.
-	AllowEntitySignedNodes bool `json:"allow_entity_signed_nodes,omitempty"`
```

Comment:
This removal actually breaks opening signed entity blobs that have explicitly set `allow_entity_signed_nodes:false`. There are some such entities on the mainnet. 

```
blob: [164 97 118 1 98 105 100 88 32 253 151 47 28 35 114 40 140 50 20 65 116 239 149 245 194 161 70 132 172 68 133 135 137 69 208 237 163 38 147 11 252 101 110 111 100 101 115 129 88 32 89 189 1 71 226 33 239 43 152 128 47 166 226 209 106 180 243 252 27 136 195 31 191 98 37 166 36 128 220 122 13 142 120 25 97 108 108 111 119 95 101 110 116 105 116 121 95 115 105 103 110 101 100 95 110 111 100 101 115 244]
deserialization into a map:
map[allow_entity_signed_nodes:false id:[253 151 47 28 35 114 40 140 50 20 65 116 239 149 245 194 161 70 132 172 68 133 135 137 69 208 237 163 38 147 11 252] nodes:[[89 189 1 71 226 33 239 43 152 128 47 166 226 209 106 180 243 252 27 136 195 31 191 98 37 166 36 128 220 122 13 142]] v:1]

deserialization into entity error:
msg="failed to fix genesis document" err="unable to open signed entity: cbor: found unknown field at map element index 3"
```

Since this is a breaking change, should probably increase descriptor version and support both versions. Note: no other special logic for old version should be needed, as on any real world network entity signed nodes are not allowed - so if set, the field will always be false.

Reply:
Ouch, yeah we should bump the version and support both (version 1 only at genesis). @abukosek can you prepare a PR?

Reply:
Done: https://github.com/oasisprotocol/oasis-core/pull/3748 :)